src/generator/AutoRest.Go/Model/MethodGo.cs

```diff
@@ -396,5 +411,99 @@ public string ValidationError()
         {
             return $"validation.NewErrorWithValidationError(err, \"{PackageName}.{Owner}\",\"{ScopedName}\")";
         }
+
+        /// <summary>
+        /// Check if method has a return response.
+        /// </summary>
+        /// <returns></returns>
+        public bool HasReturnValue()
+        {
+            return ReturnValue() != null && ReturnValue().Body != null;
+        }
+
+        /// <summary>
+        /// Return response object for the method.
+        /// </summary>
+        /// <returns></returns>
+        public Response ReturnValue()
+        {
+            return ReturnType != null
+                    ? ReturnType
+                    : DefaultResponse;
+        }
+
+        /// <summary>
+        /// Checks if method has pageable extension (x-ms-pageable) enabled.  
+        /// </summary>
+        /// <returns></returns>
+
+        public bool IsPageable()
+        {
+            return !string.IsNullOrEmpty(NextLink());
+        }
+
+        /// <summary>
+        /// Checks if method for next page of results on paged methods is already present in the method list.
+        /// </summary>
+        /// <param name="methods"></param>
+        /// <returns></returns>
+        public bool NextMethodExists(IEnumerable<MethodGo> methods)
+        {
+            if (Extensions.ContainsKey(AzureExtensions.PageableExtension))
+            {
+                var pageableExtension = JsonConvert.DeserializeObject<PageableExtension>(Extensions[AzureExtensions.PageableExtension].ToString());
+                if (pageableExtension != null && !string.IsNullOrWhiteSpace(pageableExtension.OperationName))
+                {
+                    return methods.Any(m => m.SerializedName.EqualsIgnoreCase(pageableExtension.OperationName));
+                }
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Check if method has long running extension (x-ms-long-running-operation) enabled. 
+        /// </summary>
+        /// <returns></returns>
+        public bool IsLongRunningOperation()
```

Comment:
Not for this PR, but maybe in the future... Looking at the other generators, this is the kind of func that would better live in the core, thoughts? @fearthecowboy